src/test/scala/scalaz/zio/interop/reactiveStreams/SinkToSubscriberTest.scala

```diff
@@ -2,24 +2,169 @@ package zio.interop.reactiveStreams
+              } { sbv =>
+                UIO(sbv.shutdownPublisherExecutorService())
+              }
+              .use { sbv =>
+                Task(f(sbv)).timeout(env.defaultTimeoutMillis().millis).unit
+              }
+          }
+          .catchAll {
+            case _: TestException                                                    => UIO.unit
+            case e: NullPointerException if (e.getMessage().contains("was null in")) => UIO.unit
+            case e                                                                   => IO.fail(e)
+          }
+      )
+    }
+
+  runTest(
+    "required_spec201_blackbox_mustSignalDemandViaSubscriptionRequest",
+    _.required_spec201_blackbox_mustSignalDemandViaSubscriptionRequest
+  )
+  runTest(
+    "required_spec203_blackbox_mustNotCallMethodsOnSubscriptionOrPublisherInOnComplete",
+    _.required_spec203_blackbox_mustNotCallMethodsOnSubscriptionOrPublisherInOnComplete
+  )
+  runTest(
+    "required_spec203_blackbox_mustNotCallMethodsOnSubscriptionOrPublisherInOnError",
+    _.required_spec203_blackbox_mustNotCallMethodsOnSubscriptionOrPublisherInOnError
+  )
+  runTest(
+    "required_spec205_blackbox_mustCallSubscriptionCancelIfItAlreadyHasAnSubscriptionAndReceivesAnotherOnSubscribeSignal",
+    _.required_spec205_blackbox_mustCallSubscriptionCancelIfItAlreadyHasAnSubscriptionAndReceivesAnotherOnSubscribeSignal
+  )
+  runTest(
+    "required_spec209_blackbox_mustBePreparedToReceiveAnOnCompleteSignalWithPrecedingRequestCall",
+    _.required_spec209_blackbox_mustBePreparedToReceiveAnOnCompleteSignalWithPrecedingRequestCall
+  )
+  runTest(
+    "required_spec209_blackbox_mustBePreparedToReceiveAnOnCompleteSignalWithoutPrecedingRequestCall",
+    _.required_spec209_blackbox_mustBePreparedToReceiveAnOnCompleteSignalWithoutPrecedingRequestCall
+  )
+  runTest(
+    "required_spec210_blackbox_mustBePreparedToReceiveAnOnErrorSignalWithPrecedingRequestCall",
+    _.required_spec210_blackbox_mustBePreparedToReceiveAnOnErrorSignalWithPrecedingRequestCall
+  )
+  runTest(
+    "required_spec210_blackbox_mustBePreparedToReceiveAnOnErrorSignalWithoutPrecedingRequestCall",
+    _.required_spec210_blackbox_mustBePreparedToReceiveAnOnErrorSignalWithoutPrecedingRequestCall
+  )
+  runTest(
+    "required_spec213_blackbox_onError_mustThrowNullPointerExceptionWhenParametersAreNull",
+    _.required_spec213_blackbox_onError_mustThrowNullPointerExceptionWhenParametersAreNull
+  )
+  runTest(
+    "required_spec213_blackbox_onNext_mustThrowNullPointerExceptionWhenParametersAreNull",
+    _.required_spec213_blackbox_onNext_mustThrowNullPointerExceptionWhenParametersAreNull
+  )
+  runTest(
+    "required_spec213_blackbox_onSubscribe_mustThrowNullPointerExceptionWhenParametersAreNull",
+    _.required_spec213_blackbox_onSubscribe_mustThrowNullPointerExceptionWhenParametersAreNull
+  )
+  ignore("untested_spec202_blackbox_shouldAsynchronouslyDispatch") {}
```

Comment:
Curious, why are these `ignore`? And why does the TCK specify them as untested?

Reply:
They are ignore, because the TestNG test also ignores them. In TestNG you do that by throwing a SkipException. I preferred directly ignoring them to writing another wrapper just catching the exception. They are untested in the TCK, because they cannot easily be tested in general. Part of the changes between TCK versions is additional test coverage. I guess in the TCK they are just added as placeholders for the future.